pycodestyle issues as of 2020-02

these style-issues are minor, but could be worked on slowly towards the next release.

> $ pycodestyle allantools/*.py
allantools/allantools.py:15:80: E501 line too long (88 > 79 characters)
allantools/allantools.py:112:1: E265 block comment should start with '# '
allantools/allantools.py:113:1: E265 block comment should start with '# '
allantools/allantools.py:1183:9: E722 do not use bare except'
allantools/allantools.py:1370:38: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
allantools/allantools.py:1376:37: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
allantools/allantools.py:1399:1: E302 expected 2 blank lines, found 1
allantools/allantools.py:1399:32: E251 unexpected spaces around keyword / parameter equals
allantools/allantools.py:1446:80: E501 line too long (80 > 79 characters)
allantools/allantools.py:1449:38: E261 at least two spaces before inline comment
allantools/allantools.py:1450:15: E225 missing whitespace around operator
allantools/allantools.py:1451:29: E261 at least two spaces before inline comment
allantools/allantools.py:1451:80: E501 line too long (88 > 79 characters)
allantools/allantools.py:1452:15: E225 missing whitespace around operator
allantools/allantools.py:1456:37: E261 at least two spaces before inline comment
allantools/allantools.py:1457:23: E225 missing whitespace around operator
allantools/allantools.py:1458:8: E225 missing whitespace around operator
allantools/allantools.py:1459:7: E225 missing whitespace around operator
allantools/allantools.py:1460:6: E225 missing whitespace around operator
allantools/allantools.py:1460:33: E225 missing whitespace around operator
allantools/allantools.py:1461:8: E225 missing whitespace around operator
allantools/allantools.py:1462:6: E225 missing whitespace around operator
allantools/allantools.py:1464:9: E225 missing whitespace around operator
allantools/allantools.py:1464:21: E261 at least two spaces before inline comment
allantools/allantools.py:1465:6: E225 missing whitespace around operator
allantools/allantools.py:1467:80: E501 line too long (96 > 79 characters)
allantools/allantools.py:1470:14: E225 missing whitespace around operator
allantools/allantools.py:1477:17: E225 missing whitespace around operator
allantools/allantools.py:1478:33: E261 at least two spaces before inline comment
allantools/allantools.py:1479:17: E225 missing whitespace around operator
allantools/allantools.py:1481:14: E225 missing whitespace around operator
allantools/allantools.py:1482:11: E128 continuation line under-indented for visual indent
allantools/allantools.py:1483:11: E128 continuation line under-indented for visual indent
allantools/allantools.py:1485:14: E225 missing whitespace around operator
allantools/allantools.py:1486:6: E225 missing whitespace around operator
allantools/allantools.py:1487:23: E225 missing whitespace around operator
allantools/ci.py:108:80: E501 line too long (88 > 79 characters)
allantools/ci.py:143:80: E501 line too long (85 > 79 characters)
allantools/ci.py:145:80: E501 line too long (87 > 79 characters)
allantools/ci.py:203:80: E501 line too long (82 > 79 characters)
allantools/ci.py:236:80: E501 line too long (94 > 79 characters)
allantools/ci.py:376:80: E501 line too long (84 > 79 characters)
allantools/ci.py:387:80: E501 line too long (102 > 79 characters)
allantools/ci.py:443:16: E127 continuation line over-indented for visual indent
allantools/ci.py:447:80: E501 line too long (84 > 79 characters)
allantools/ci.py:480:80: E501 line too long (84 > 79 characters)
allantools/ci.py:519:24: E127 continuation line over-indented for visual indent
allantools/ci.py:534:24: E127 continuation line over-indented for visual indent
allantools/ci.py:546:80: E501 line too long (87 > 79 characters)
allantools/ci.py:547:24: E127 continuation line over-indented for visual indent
allantools/ci.py:547:80: E501 line too long (88 > 79 characters)
allantools/ci.py:555:80: E501 line too long (92 > 79 characters)
allantools/ci.py:570:24: E127 continuation line over-indented for visual indent
allantools/ci.py:578:80: E501 line too long (80 > 79 characters)
allantools/ci.py:580:80: E501 line too long (84 > 79 characters)
allantools/ci.py:594:80: E501 line too long (87 > 79 characters)
allantools/ci.py:597:80: E501 line too long (95 > 79 characters)
allantools/ci.py:684:80: E501 line too long (89 > 79 characters)
allantools/ci.py:686:80: E501 line too long (86 > 79 characters)
allantools/ci.py:691:14: E124 closing bracket does not match visual indentation
allantools/ci.py:700:80: E501 line too long (89 > 79 characters)
allantools/ci.py:707:14: E124 closing bracket does not match visual indentation
allantools/ci.py:736:80: E501 line too long (93 > 79 characters)
allantools/ci.py:808:80: E501 line too long (83 > 79 characters)
allantools/mask.py:4:1: W293 blank line contains whitespace
allantools/mask.py:10:1: W293 blank line contains whitespace
allantools/mask.py:13:1: W293 blank line contains whitespace
allantools/mask.py:18:1: E302 expected 2 blank lines, found 1
allantools/mask.py:19:48: W291 trailing whitespace
allantools/mask.py:24:11: E225 missing whitespace around operator
allantools/mask.py:26:11: E225 missing whitespace around operator
allantools/mask.py:31:1: E302 expected 2 blank lines, found 1
allantools/mask.py:32:38: W291 trailing whitespace
allantools/mask.py:37:11: E225 missing whitespace around operator
allantools/mask.py:42:1: E302 expected 2 blank lines, found 1
allantools/mask.py:43:62: W291 trailing whitespace
allantools/mask.py:50:11: E225 missing whitespace around operator
allantools/mask.py:52:11: E225 missing whitespace around operator
allantools/mask.py:57:1: E302 expected 2 blank lines, found 1
allantools/mask.py:63:1: W293 blank line contains whitespace
allantools/mask.py:66:11: E225 missing whitespace around operator
allantools/mask.py:70:11: E225 missing whitespace around operator
allantools/mask.py:73:23: E261 at least two spaces before inline comment
allantools/mask.py:74:1: W293 blank line contains whitespace
allantools/mask.py:75:1: E302 expected 2 blank lines, found 1
allantools/mask.py:81:1: W293 blank line contains whitespace
allantools/mask.py:84:11: E225 missing whitespace around operator
allantools/mask.py:86:11: E225 missing whitespace around operator
allantools/mask.py:91:1: E302 expected 2 blank lines, found 1
allantools/mask.py:97:1: W293 blank line contains whitespace
allantools/mask.py:100:11: E225 missing whitespace around operator
allantools/mask.py:102:11: E225 missing whitespace around operator
allantools/mask.py:106:1: W293 blank line contains whitespace
allantools/mask.py:107:1: E302 expected 2 blank lines, found 1
allantools/mask.py:111:1: W293 blank line contains whitespace
allantools/mask.py:114:11: E225 missing whitespace around operator
allantools/mask.py:115:20: E225 missing whitespace around operator
allantools/mask.py:117:24: E261 at least two spaces before inline comment
allantools/mask.py:117:32: W291 trailing whitespace
allantools/mask.py:119:1: W293 blank line contains whitespace
allantools/mask.py:124:1: W293 blank line contains whitespace
allantools/mask.py:127:11: E225 missing whitespace around operator
allantools/mask.py:128:20: E225 missing whitespace around operator
allantools/mask.py:130:25: E261 at least two spaces before inline comment
allantools/mask.py:130:32: W291 trailing whitespace
allantools/mask.py:131:1: W293 blank line contains whitespace
allantools/mask.py:131:1: W391 blank line at end of file
allantools/noise_kasdin.py:244:80: E501 line too long (100 > 79 characters)
allantools/noise_kasdin.py:289:80: E501 line too long (93 > 79 characters)
allantools/noise_kasdin.py:293:80: E501 line too long (92 > 79 characters)
allantools/noise.py:110:80: E501 line too long (80 > 79 characters)
allantools/noise.py:134:80: E501 line too long (85 > 79 characters)
allantools/noise.py:136:1: E302 expected 2 blank lines, found 1
allantools/noise.py:156:80: E501 line too long (80 > 79 characters)
allantools/noise.py:205:1: W391 blank line at end of file
allantools/plot.py:70:13: E124 closing bracket does not match visual indentation
allantools/plot.py:91:29: E124 closing bracket does not match visual indentation
allantools/plot.py:96:25: E124 closing bracket does not match visual indentation
allantools/realtime.py:44:80: E501 line too long (84 > 79 characters)
allantools/realtime.py:62:80: E501 line too long (103 > 79 characters)
allantools/realtime.py:66:80: E501 line too long (85 > 79 characters)
allantools/realtime.py:93:80: E501 line too long (86 > 79 characters)
allantools/realtime.py:99:80: E501 line too long (114 > 79 characters)
allantools/realtime.py:117:80: E501 line too long (91 > 79 characters)
allantools/realtime.py:124:80: E501 line too long (89 > 79 characters)
allantools/realtime.py:131:80: E501 line too long (114 > 79 characters)
allantools/realtime.py:151:80: E501 line too long (86 > 79 characters)
allantools/realtime.py:153:80: E501 line too long (95 > 79 characters)
allantools/realtime.py:157:80: E501 line too long (102 > 79 characters)
allantools/realtime.py:163:80: E501 line too long (113 > 79 characters)
allantools/realtime.py:197:80: E501 line too long (102 > 79 characters)
allantools/realtime.py:205:80: E501 line too long (81 > 79 characters)
allantools/realtime.py:209:80: E501 line too long (96 > 79 characters)

--- a/.gitignore
+++ b/.gitignore
@@ -13,3 +13,4 @@ dist/**
 *.egg
 .coverage
 htmlcov/
+.python-version

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,12 +6,25 @@
 matrix:
     include:
         - os: linux
-          name: "Linux/xenial - Python 2.7, with coverage"
+          name: "Linux/xenial - Python 2.7"
           dist: xenial   # required for Python >= 3.7
           language: python
           sudo: required
           python:        # removed 3.3 because scipy gives and error
-              - "2.7"
+              - 2.7
+          install:
+              - sudo apt-get install liblapack-dev libblas-dev gfortran
+              - pip install -v .      # allantools itself, verbose output
+              - pip install -r requirements.txt
+          script:
+              - py.test
+        - os: linux
+          name: "Linux/xenial - Python 3.4, with coverage"
+          dist: xenial   # required for Python >= 3.7
+          language: python
+          sudo: required
+          python:        # removed 3.3 because scipy gives and error
+              - 3.4
           install:
               - sudo apt-get install liblapack-dev libblas-dev gfortran
               - pip install -v .      # allantools itself, verbose output
@@ -21,19 +34,6 @@ matrix:
               - py.test
               - coverage run --source=allantools setup.py test
           after_success: coveralls
-        - os: linux
-          name: "Linux/xenial - Python 3.4"
-          dist: xenial   # required for Python >= 3.7
-          language: python
-          sudo: required
-          python:        # removed 3.3 because scipy gives and error
-              - "3.4"
-          install:
-              - sudo apt-get install liblapack-dev libblas-dev gfortran
-              - pip install -v .      # allantools itself, verbose output
-              - pip install -r requirements.txt
-          script:
-              - py.test
         - os: osx           # https://docs.travis-ci.com/user/reference/osx/
           name: "MacOS 10.14.3 - Python 3.7.2 - experimental"
           osx_image: xcode10.2  # Python 3.7.2 running on macOS 10.14.3
@@ -46,12 +46,14 @@ matrix:
         - os: windows
           name: "Windows - Python 3.8 - experimental"
           language: shell
-          env: PATH=/c/Python:/c/Python/Scripts:$PATH
+          env: PATH=/c/Python38:/c/Python38/Scripts:$PATH
           before_install:
-              - choco install python3 --version 3.8.0
+              - choco install python --version 3.8.0
+              - python --version
               - python -m ensurepip
-              - pip install -v .
-              - pip install -r requirements.txt
+          install:
+              - python -m pip install -v .
+              - python -m pip install -r requirements.txt
           script:
               - py.test
 # end

--- a/.travis.yml
+++ b/.travis.yml
@@ -46,11 +46,12 @@ matrix:
         - os: windows
           name: "Windows - Python 3.8 - experimental"
           language: shell
+          env: PATH=/c/Python:/c/Python/Scripts:$PATH
           before_install:
-              - choco install python --version 3.8.0
+              - choco install python3 --version 3.8.0
+              - python -m ensurepip
               - pip install -v .
               - pip install -r requirements.txt
-          env: PATH=/c/Python37:/c/Python37/Scripts:$PATH
           script:
               - py.test
 # end

--- a/.travis.yml
+++ b/.travis.yml
@@ -44,7 +44,7 @@ matrix:
           script:
               - py.test
         - os: windows
-          name: "Windows - Python 3.7 - experimental"
+          name: "Windows - Python 3.8 - experimental"
           language: shell
           before_install:
               - choco install python --version 3.8.0

--- a/allantools/allantools.py
+++ b/allantools/allantools.py
@@ -12,7 +12,8 @@ Version history
 - psd2allan() - convert PSD to ADEV/MDEV
 
 **2019.09** 2019 September
-- packaging changes, for conda package (see https://anaconda.org/conda-forge/allantools)
+- packaging changes, for conda package
+  (see https://anaconda.org/conda-forge/allantools)
 
 **2019.07** 2019 August 3
 - move edf-functions and noise-ID functions to ci.py
@@ -109,8 +110,8 @@ import numpy as np
 
 from scipy import interpolate
 from scipy.integrate import simps
-#import scipy.stats # used in confidence_intervals()
-#import scipy.signal # decimation in lag-1 acf
+# import scipy.stats # used in confidence_intervals()
+# import scipy.signal # decimation in lag-1 acf
 
 
 from . import ci  # edf, confidence intervals
@@ -761,7 +762,6 @@ def calc_mtotdev_phase(phase, rate, m):
 
     n = 0      # number of terms in the sum, for error estimation
     dev = 0.0  # the deviation we are computing
-    err = 0.0  # the error in the deviation
     # print('calc_mtotdev N=%d m=%d' % (N,m) )
     for i in range(0, N-3*m+1):
         # subsequence of length 3m, from the original phase data
@@ -1367,15 +1367,17 @@ def calc_gradev_phase(data, rate, mj, stride, confidence, noisetype):
 
     v_arr = d2[:n] - 2 * d1[:n] + d0[:n]
 
-    n = len(np.where(np.isnan(v_arr) == False)[0])  # only average for non-nans
+    # only average for non-nans
+    n = len(np.where(np.isnan(v_arr) == False)[0])  # noqa
 
     if n == 0:
         RuntimeWarning("Data array length is too small: %i" % len(data))
         n = 1
 
-    N = len(np.where(np.isnan(data) == False)[0])
+    N = len(np.where(np.isnan(data) == False)[0])  # noqa
 
-    s = np.nansum(v_arr * v_arr)   # a summation robust to nans
+    # a summation robust to nans
+    s = np.nansum(v_arr * v_arr)
 
     dev = np.sqrt(s / (2.0 * n)) / mj * rate
     # deverr = dev / np.sqrt(n) # old simple errorbars
@@ -1396,7 +1398,8 @@ def calc_gradev_phase(data, rate, mj, stride, confidence, noisetype):
 
     return dev, deverr, n
 
-def psd2allan(S_y, f=1.0, kind= 'adev', base=2):
+
+def psd2allan(S_y, f=1.0, kind='adev', base=2):
     """ Convert a given (one-sided) power spectral density S_y(f) to Allan
         deviation or modified Allan deviation
 
@@ -1443,48 +1446,51 @@ def psd2allan(S_y, f=1.0, kind= 'adev', base=2):
         Computed Allan deviation of requested kind for each tau value
     """
     # determine taus from df
-    # first oversample S_y by a factor of 10 in order to avoid numerical problem
-    # at tau > 1/2df
+    # first oversample S_y by a factor of 10 in order to avoid numerical
+    # problem at tau > 1/2df
     if isinstance(S_y, np.ndarray):
-        if isinstance(f, np.ndarray): # f is frequency vector
-            df= f[1]-f[0]
-        elif np.isscalar(f): # assume that f is the frequency step, not frequency vector
-            df= f
+        if isinstance(f, np.ndarray):  # f is frequency vector
+            df = f[1]-f[0]
+        elif np.isscalar(f):
+            # assume that f is the frequency step, not frequency vector
+            df = f
         else:
             raise ValueError(np.ndarray, float, int)
     else:
-        raise ValueError(np.ndarray) # raise error
-    oversamplingfactor= 4
-    df0= oversamplingfactor * df
-    f0= np.arange(S_y.size * df0, step=df0)
-    f= np.arange(df, (S_y.size -1) * df0 + df, df)
-    S_y= interpolate.interp1d(f0, S_y, kind='cubic')(f)
-    f= f / oversamplingfactor
+        raise ValueError(np.ndarray)  # raise error
+    oversamplingfactor = 4
+    df0 = oversamplingfactor * df
+    f0 = np.arange(S_y.size * df0, step=df0)
+    f = np.arange(df, (S_y.size - 1) * df0 + df, df)
+    S_y = interpolate.interp1d(f0, S_y, kind='cubic')(f)
+    f = f / oversamplingfactor
 
-    tau0=1/np.max(f) # minimum tau derived from the given frequency vector
-    n=1/df/tau0/2
+    tau0 = 1/np.max(f)  # minimum tau derived from the given frequency vector
+    n = 1/df/tau0/2
     if base > 1:
-        m = np.unique(np.round(np.append(base**np.arange(np.floor(np.log(n)/np.log(base))), n)))
+        m = np.unique(np.round(np.append(base**np.arange(
+            np.floor(np.log(n)/np.log(base))), n)))
     else:
         m = np.arange(1, n)
-    taus_used= m*tau0
+    taus_used = m*tau0
 
     # TODO: In principle, this approach can be extended to the other kinds of
     # Allan deviations, we just need to determine the respective transfer
     # function in the frequency domain.
 
     if kind[0].lower() == 'a':   # for ADEV
-        exponent=1.0
-    elif kind[0].lower() == 'm': # for modADEV
-        exponent=2.0
+        exponent = 1.0
+    elif kind[0].lower() == 'm':  # for modADEV
+        exponent = 2.0
 
-    integrand= np.array([S_y *
-          np.abs(np.sin(np.pi * f * taus_used[idx])**(exponent + 1.0)
-          / (np.pi * f * taus_used[idx])**exponent)**2.0
-          for idx, mj in enumerate(m)])
-    integrand= np.insert(integrand, 0, 0.0, axis=1)
-    f= np.insert(f, 0, 0.0)
-    ad = np.sqrt(2.0 *simps(integrand, f))
+    integrand = np.array([
+        S_y *
+        np.abs(np.sin(np.pi * f * taus_used[idx])**(exponent + 1.0)
+               / (np.pi * f * taus_used[idx])**exponent)**2.0
+        for idx, mj in enumerate(m)])
+    integrand = np.insert(integrand, 0, 0.0, axis=1)
+    f = np.insert(f, 0, 0.0)
+    ad = np.sqrt(2.0 * simps(integrand, f))
     return taus_used, ad
 
 

--- a/allantools/ci.py
+++ b/allantools/ci.py
@@ -28,6 +28,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 import numpy as np
 import scipy.stats  # used in confidence_intervals()
 import scipy.signal  # decimation in lag-1 acf
+import allantools as at
 
 ########################################################################
 # Confidence Intervals
@@ -105,7 +106,8 @@ def confidence_interval_noiseID(x, dev, af, dev_type="adev",
     dmax = 2
     if (dev_type is "hdev") or (dev_type is "ohdev"):
         dmax = 3
-    alpha_int = autocorr_noise_id(x, int(af), data_type=data_type, dmin=0, dmax=dmax)[0]
+    alpha_int = autocorr_noise_id(
+        x, int(af), data_type=data_type, dmin=0, dmax=dmax)[0]
 
     # 2) EDF
     if dev_type is "adev":
@@ -140,9 +142,11 @@ def rn(x, af, rate):
 
         ratio of MVAR to AVAR
     """
-    (taus, devs, errs, ns) = at.adev(x, taus=[af*rate], data_type='phase', rate=rate)
+    (taus, devs, errs, ns) = at.adev(
+        x, taus=[af*rate], data_type='phase', rate=rate)
     oadev_x = devs[0]
-    (mtaus, mdevs, errs, ns) = at.mdev(x, taus=[af*rate], data_type='phase', rate=rate)
+    (mtaus, mdevs, errs, ns) = at.mdev(
+        x, taus=[af*rate], data_type='phase', rate=rate)
     mdev_x = mdevs[0]
     return pow(mdev_x/oadev_x, 2)
 
@@ -180,7 +184,7 @@ def rn_boundary(af, b_hi):
     R(n) ratio boundary for selecting between [b_hi-1, b_hi]
     alpha = b + 2
     """
-    return np.sqrt(rn_theory(af, b)*rn_theory(af, b-1))  # geometric mean
+    return np.sqrt(rn_theory(af, b_hi)*rn_theory(af, b_hi-1))  # geometric mean
 
 ########################################################################
 # Noise Identification using B1
@@ -200,7 +204,8 @@ def b1(x, af, rate):
         Barnes, 1974
         https://tf.nist.gov/general/pdf/11.pdf
     """
-    (taus, devs, errs, ns) = adev(x, taus=[af*rate], data_type="phase", rate=rate)
+    (taus, devs, errs, ns) = at.adev(
+        x, taus=[af*rate], data_type="phase", rate=rate)
     oadev_x = devs[0]
     avar = pow(oadev_x, 2.0)
 
@@ -233,7 +238,8 @@ def b1_theory(N, mu):
        -3    -1       0
        -4    -2      +1
        -5    -3      +2
-       -6    -4      +3 for HDEV, by applying B1 to frequency data, and add +2 to resulting mu
+       -6    -4      +3 for HDEV, by applying B1 to frequency data,
+                        and add +2 to resulting mu
     """
 
     # see Table 3 of Howe 2000
@@ -368,12 +374,11 @@ def autocorr_noise_id(x, af, data_type="phase", dmin=0, dmax=2):
 
     """
     d = 0  # number of differentiations
-    lag = 1
     if data_type is "phase":
         if af > 1:
             # x = scipy.signal.decimate(x, af, n=1, ftype='fir')
             x = x[0:len(x):af]  # decimate by averaging factor
-        x = detrend(x, deg=2)  # remove quadratic trend (frequency offset and drift)
+        x = detrend(x, deg=2)  # remove quadratic trend (freq offset and drift)
     elif data_type is "freq":
         # average by averaging factor
         y_cut = np.array(x[:len(x)-(len(x) % af)])  # cut to length
@@ -384,7 +389,8 @@ def autocorr_noise_id(x, af, data_type="phase", dmin=0, dmax=2):
 
     # require minimum length for time-series
     if len(x) < 30:
-        print("autocorr_noise_id() Don't know how to do noise-ID for time-series length= %d" % len(x))
+        print(("autocorr_noise_id() Don't know how to do noise-ID for"
+               " time-series length= %d") % len(x))
         raise NotImplementedError
 
     while True:
@@ -439,12 +445,13 @@ def edf_greenhall_simple(alpha, d, m, S, F, N):
     L = m/F+m*d  # length of filter applied to phase samples
     M = 1 + np.floor(S*(N-L) / m)
     J = min(M, (d+1)*S)
-    inv_edf = (1.0/(pow(greenhall_sz(0, F, alpha, d), 2)*M)) * \
-               greenhall_BasicSum(J, M, S, F, alpha, d)
+    inv_edf = ((1.0/(pow(greenhall_sz(0, F, alpha, d), 2)*M)) *
+               greenhall_BasicSum(J, M, S, F, alpha, d))
     return 1.0/inv_edf
 
 
-def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=False):
+def edf_greenhall(alpha, d, m, N,
+                  overlapping=False, modified=False, verbose=False):
     """ returns Equivalent degrees of freedom
 
         Parameters
@@ -477,7 +484,8 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
 
         Notes
         -----
-        Used for the following deviations (see http://www.wriley.com/CI2.pdf page 8)
+        Used for the following deviations
+        (see http://www.wriley.com/CI2.pdf page 8)
         adev()
         oadev()
         mdev()
@@ -515,8 +523,8 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
             return 1.0/inv_edf
         else:
             m_prime = J_max/r
-            inv_edf = (1.0/(pow(greenhall_sz(0, F, alpha, d), 2)*J_max)) * \
-                       greenhall_BasicSum(J_max, J_max, m_prime, 1, alpha, d)
+            inv_edf = ((1.0/(pow(greenhall_sz(0, F, alpha, d), 2)*J_max)) *
+                       greenhall_BasicSum(J_max, J_max, m_prime, 1, alpha, d))
             if verbose:
                 print("case 1.3 edf= %3f" % float(1.0/inv_edf))
             return 1.0/inv_edf
@@ -530,8 +538,8 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
                 m_prime = float('inf')
                 variant = "b"
 
-            inv_edf = (1.0/(pow(greenhall_sz(0, m_prime, alpha, d), 2)*M)) * \
-                       greenhall_BasicSum(J, M, S, m_prime, alpha, d)
+            inv_edf = ((1.0/(pow(greenhall_sz(0, m_prime, alpha, d), 2)*M)) *
+                       greenhall_BasicSum(J, M, S, m_prime, alpha, d))
             if verbose:
                 print("case 2.1%s edf= %3f" % (variant, float(1.0/inv_edf)))
             return 1.0/inv_edf
@@ -543,16 +551,19 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
             return 1.0/inv_edf
         else:
             m_prime = J_max/r
-            inv_edf = (1.0/(pow(greenhall_sz(0, float('inf'), alpha, d), 2)*J_max)) * \
-                       greenhall_BasicSum(J_max, J_max, m_prime, float('inf'), alpha, d)
+            inv_edf = (
+                (1.0/(pow(greenhall_sz(0, float('inf'), alpha, d), 2)*J_max)) *
+                greenhall_BasicSum(
+                    J_max, J_max, m_prime, float('inf'), alpha, d))
             if verbose:
                 print("case 2.3 edf= %3f" % float(1.0/inv_edf))
             return 1.0/inv_edf
     elif int(F) == int(m) and int(alpha) == 1 and not modified:
         # case 3, unmodified variances, alpha=1
         if J <= J_max:
-            inv_edf = (1.0/(pow(greenhall_sz(0, m, 1, d), 2)*M)) * \
-                       greenhall_BasicSum(J, M, S, m, 1, d)  # note: m<1e6 to avoid roundoff
+            # note: m<1e6 to avoid roundoff
+            inv_edf = ((1.0/(pow(greenhall_sz(0, m, 1, d), 2)*M)) *
+                       greenhall_BasicSum(J, M, S, m, 1, d))
             if verbose:
                 print("case 3.1 edf= %3f" % float(1.0/inv_edf))
             return 1.0/inv_edf
@@ -566,8 +577,9 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
         else:
             m_prime = J_max/r
             (b0, b1) = greenhall_table3(alpha, d)
-            inv_edf = (1.0/(pow(b0+b1*np.log(m), 2)*J_max)) * \
-                       greenhall_BasicSum(J_max, J_max, m_prime, m_prime, 1, d)
+            inv_edf = (
+                (1.0/(pow(b0+b1*np.log(m), 2)*J_max)) *
+                greenhall_BasicSum(J_max, J_max, m_prime, m_prime, 1, d))
             if verbose:
                 print("case 3.3 edf= %3f" % float(1.0/inv_edf))
             return 1.0/inv_edf
@@ -575,9 +587,11 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
         # case 4, unmodified variances, alpha=2
         K = np.ceil(r)
         if K <= d:
-            raise NotImplementedError  # FIXME: add formula from the paper here!
+            # FIXME: add formula from the paper here!
+            raise NotImplementedError
         else:
-            a0 = scipy.special.binom(4*d, 2*d) / pow(scipy.special.binom(2*d, d), 2)
+            a0 = (scipy.special.binom(4*d, 2*d) /
+                  pow(scipy.special.binom(2*d, d), 2))
             a1 = d/2.0
             inv_edf = (1.0/M)*(a0-a1/r)
             if verbose:
@@ -591,10 +605,12 @@ def edf_greenhall(alpha, d, m, N, overlapping=False, modified=False, verbose=Fal
 def greenhall_BasicSum(J, M, S, F, alpha, d):
     """ Eqn (10) from Greenhall2004 """
     first = pow(greenhall_sz(0, F, alpha, d), 2)
-    second = (1-float(J)/float(M))*pow(greenhall_sz(float(J)/float(S), F, alpha, d), 2)
+    second = ((1-float(J)/float(M)) *
+              pow(greenhall_sz(float(J)/float(S), F, alpha, d), 2))
     third = 0
     for j in range(1, int(J)):
-        third += 2*(1.0-float(j)/float(M))*pow(greenhall_sz(float(j)/float(S), F, alpha, d), 2)
+        third += (2 * (1.0-float(j)/float(M)) *
+                  pow(greenhall_sz(float(j) / float(S), F, alpha, d), 2))
     return first+second+third
 
 
@@ -681,14 +697,21 @@ def greenhall_table2(alpha, d):
     row_idx = int(-alpha+2)  # map 2-> row0 and -4-> row6
     assert(row_idx in [0, 1, 2, 3, 4, 5])
     col_idx = int(d-1)
-    table2 = [[(3.0/2.0, 1.0/2.0), (35.0/18.0, 1.0), (231.0/100.0, 3.0/2.0)],  # alpha=+2
-              [(78.6, 25.2), (790.0, 410.0), (9950.0, 6520.0)],
-              [(2.0/3.0, 1.0/6.0), (2.0/3.0, 1.0/3.0), (7.0/9.0, 1.0/2.0)],  # alpha=0
-              [(-1, -1), (0.852, 0.375), (0.997, 0.617)],  # -1
-              [(-1, -1), (1.079, 0.368), (1.033, 0.607)],  # -2
-              [(-1, -1), (-1, -1), (1.053, 0.553)],  # -3
-              [(-1, -1), (-1, -1), (1.302, 0.535)],  # alpha=-4
-             ]
+    table2 = [
+        # alpha = +2:
+        [(3.0/2.0, 1.0/2.0), (35.0/18.0, 1.0), (231.0/100.0, 3.0/2.0)],
+        # alpha = +1:
+        [(78.6, 25.2), (790.0, 410.0), (9950.0, 6520.0)],
+        # alpha = 0:
+        [(2.0/3.0, 1.0/6.0), (2.0/3.0, 1.0/3.0), (7.0/9.0, 1.0/2.0)],
+        # alpha = -1:
+        [(-1, -1), (0.852, 0.375), (0.997, 0.617)],
+        # alpha = -2
+        [(-1, -1), (1.079, 0.368), (1.033, 0.607)],
+        # alpha = -3
+        [(-1, -1), (-1, -1), (1.053, 0.553)],
+        # alpha = -4
+        [(-1, -1), (-1, -1), (1.302, 0.535)]]
     # print("table2 = ", table2[row_idx][col_idx])
     return table2[row_idx][col_idx]
 
@@ -697,14 +720,21 @@ def greenhall_table1(alpha, d):
     """ Table 1 from Greenhall 2004 """
     row_idx = int(-alpha+2)  # map 2-> row0 and -4-> row6
     col_idx = int(d-1)
-    table1 = [[(2.0/3.0, 1.0/3.0), (7.0/9.0, 1.0/2.0), (22.0/25.0, 2.0/3.0)],  # alpha=+2
-              [(0.840, 0.345), (0.997, 0.616), (1.141, 0.843)],
-              [(1.079, 0.368), (1.033, 0.607), (1.184, 0.848)],
-              [(-1, -1), (1.048, 0.534), (1.180, 0.816)],  # -1
-              [(-1, -1), (1.302, 0.535), (1.175, 0.777)],  # -2
-              [(-1, -1), (-1, -1), (1.194, 0.703)],  # -3
-              [(-1, -1), (-1, -1), (1.489, 0.702)],  # alpha=-4
-             ]
+    table1 = [
+        # alpha = +2:
+        [(2.0/3.0, 1.0/3.0), (7.0/9.0, 1.0/2.0), (22.0/25.0, 2.0/3.0)],
+        # alpha = +1:
+        [(0.840, 0.345), (0.997, 0.616), (1.141, 0.843)],
+        # alpha = 0:
+        [(1.079, 0.368), (1.033, 0.607), (1.184, 0.848)],
+        # alpha = -1:
+        [(-1, -1), (1.048, 0.534), (1.180, 0.816)],
+        # alpha = -2
+        [(-1, -1), (1.302, 0.535), (1.175, 0.777)],
+        # alpha = -3
+        [(-1, -1), (-1, -1), (1.194, 0.703)],
+        # alpha = -4
+        [(-1, -1), (-1, -1), (1.489, 0.702)]]
     # print("table1 = ", table1[row_idx][col_idx])
     return table1[row_idx][col_idx]
 
@@ -733,7 +763,11 @@ def edf_mtotdev(N, m, alpha):
         NIST SP1065 page 41, Table 8
     """
     assert(alpha in [2, 1, 0, -1, -2])
-    NIST_SP1065_table8 = [(1.90, 2.1), (1.20, 1.40), (1.10, 1.2), (0.85, 0.50), (0.75, 0.31)]
+    NIST_SP1065_table8 = [(1.90, 2.1),
+                          (1.20, 1.40),
+                          (1.10, 1.2),
+                          (0.85, 0.50),
+                          (0.75, 0.31)]
     # (b, c) = NIST_SP1065_table8[ abs(alpha-2) ]
     (b, c) = NIST_SP1065_table8[abs(alpha-2)]
     edf = b*(float(N)/float(m))-c
@@ -805,7 +839,8 @@ def edf_simple(N, m, alpha):
 
     else:
         edf = (N - 1)
-        print("Noise type not recognized. Defaulting to N - 1 degrees of freedom.")
+        print("Noise type not recognized."
+              " Defaulting to N - 1 degrees of freedom.")
 
     return edf
 

--- a/allantools/mask.py
+++ b/allantools/mask.py
@@ -1,58 +1,62 @@
 """
     This file is part of AllanTools
     https://github.com/aewallin/allantools
-    
+
     TDEV and MTIE masks for ITU Standards
     AW2017-2018
 
     ITU Primary Reference Clock
     https://www.itu.int/rec/T-REC-G.811/en
-    
+
     ITU Primary Reference Time Clock
     https://www.itu.int/rec/T-REC-G.8272/en
-    
+
     ITU Enhanced Primary Reference Time Clock
     https://www.itu.int/rec/T-REC-G.8272.1/en
 """
 
+
 def prc_tdev(tau):
-    """ Time Deviation, ITU Primary Reference Clock 
+    """ Time Deviation, ITU Primary Reference Clock
         https://www.itu.int/rec/T-REC-G.811/en
         Recommendation G.811 (1997) Amendment 1 (04/16)
         Approved in 2016-04-13
     """
-    if tau<100:
+    if tau < 100:
         return 3e-9
-    if tau<1000:
+    if tau < 1000:
         return 0.03*1.0e-9*tau
     else:
         return 30e-9
 
+
 def prc_mtie(tau):
-    """ MTIE, ITU Primary Reference Clock 
+    """ MTIE, ITU Primary Reference Clock
         https://www.itu.int/rec/T-REC-G.811/en
         Recommendation G.811 (1997) Amendment 1 (04/16)
         Approved in 2016-04-13
     """
-    if tau<1000:
+    if tau < 1000:
         return 1e-6 * (tau*0.275e-3 + 0.025)
     else:
         return 1e-6 * (tau*1.0e-5 + 0.29)
 
+
 def eprtc_tdev(tau):
-    """ Time Deviation, ITU Enhanced Primary Reference Time Clock 
+    """ Time Deviation, ITU Enhanced Primary Reference Time Clock
         https://www.itu.int/rec/T-REC-G.8272.1/en
         Recommendation G.8272.1/Y.1367.1 (2016) Amendment 1 (08/17)
         Approved in 2017-08-13
 
         Table 2 TDEV, page 4
     """
-    if tau<30e3:
+    if tau < 30e3:
         return 1e-9
-    if tau<300e3:
+    if tau < 300e3:
         return 3.33333e-5*1.0e-9*tau
     else:
         return 10e-9
+
 
 def eprtc_mtie(tau):
     """ MTIE
@@ -60,33 +64,35 @@ def eprtc_mtie(tau):
         https://www.itu.int/rec/T-REC-G.8272.1/en
         Recommendation G.8272.1/Y.1367.1 (2016) Amendment 1 (08/17)
         Approved in 2017-08-13
-        
+
         Table 1, page 3
     """
-    if tau<=1:
+    if tau <= 1:
         return 4e-9
     if tau <= 100:
         return 1e-9*(0.11114*tau+3.89)
-    if tau<= 400000:
+    if tau <= 400000:
         return 1e-9*(0.0375e-3*tau+15)
     else:
-        return 1e-9*30 # 30 ns
-        
+        return 1e-9*30  # 30 ns
+
+
 def prtcA_tdev(tau):
     """ Time Deviation
         ITU Primary Reference Time Clock
         https://www.itu.int/rec/T-REC-G.8272/en
         Recommendation G.8272/Y.1367 (11/18)
         Approved in 2018-11-29
-        
+
         Table 3
     """
-    if tau<100:
+    if tau < 100:
         return 3e-9
-    if tau<1000:
+    if tau < 1000:
         return 0.03e-9*tau
     else:
         return 30e-9
+
 
 def prtcB_tdev(tau):
     """ Time Deviation
@@ -94,38 +100,38 @@ def prtcB_tdev(tau):
         https://www.itu.int/rec/T-REC-G.8272/en
         Recommendation G.8272/Y.1367 (11/18)
         Approved in 2018-11-29
-        
+
         Table 4
     """
-    if tau<100:
+    if tau < 100:
         return 1e-9
-    if tau<1000:
+    if tau < 1000:
         return 0.01e-9*tau
     else:
         return 5e-9
-        
+
+
 def prtcA_mtie(tau):
     """ MTIE
         ITU Primary Reference Time Clock
         https://www.itu.int/rec/T-REC-G.8272/en
-        
+
         Table 1
     """
-    if tau<273:
-        return 1e-6* (0.275e-3*tau + 0.025)
+    if tau < 273:
+        return 1e-6 * (0.275e-3*tau + 0.025)
     else:
-        return 1e-6*0.1 # 100ns     
+        return 1e-6 * 0.1  # 100ns
 
-        
+
 def prtcB_mtie(tau):
     """ MTIE
         ITU Primary Reference Time Clock
         https://www.itu.int/rec/T-REC-G.8272/en
-        
+
         Table 2
     """
-    if tau<54.5:
-        return 1e-6* (0.275e-3*tau + 0.025)
+    if tau < 54.5:
+        return 1e-6 * (0.275e-3*tau + 0.025)
     else:
-        return 1e-6*0.04 # 40ns    
-        
+        return 1e-6 * 0.04  # 40ns

--- a/allantools/noise.py
+++ b/allantools/noise.py
@@ -106,9 +106,10 @@ def brown(num_points=1024, b_minus2=1.0, fs=1.0):
         -------
         Random walk sample: numpy.array
     """
-    return (1.0/float(fs))*numpy.cumsum(white(num_points,
-                                              b0=b_minus2*(4.0*math.pi*math.pi),
-                                              fs=fs))
+    return (1.0/float(fs))*numpy.cumsum(
+        white(num_points,
+              b0=b_minus2*(4.0*math.pi*math.pi),
+              fs=fs))
 
 
 def violet(num_points=1024, b2=1, fs=1):
@@ -131,7 +132,9 @@ def violet(num_points=1024, b2=1, fs=1):
         Violet noise sample: numpy.array
     """
     # diff() reduces number of points by one.
-    return (float(fs))*numpy.diff(white(num_points+1, b0=b2/(2.0*math.pi)**2, fs=fs))
+    return (float(fs))*numpy.diff(
+        white(num_points+1, b0=b2/(2.0*math.pi)**2, fs=fs))
+
 
 def pink(num_points=1024, depth=80):
     """ Pink noise (approximation) with 1/f PSD
@@ -153,8 +156,8 @@ def pink(num_points=1024, depth=80):
         -------
         Pink noise sample: numpy.array
     """
-    # FIXME: couldn't we implement here the normalization as for the other noise
-    # types using a noise power law coefficient b_minus1?
+    # FIXME: couldn't we implement here the normalization as for the other
+    # noise types using a noise power law coefficient b_minus1?
     a = []
     s = iterpink(depth)
     for n in range(num_points):  # FIXME: num_points is unused here.

--- a/allantools/noise_kasdin.py
+++ b/allantools/noise_kasdin.py
@@ -241,7 +241,8 @@ class Noise(object):
         if self.b == 0:
             coeff = 3.0*f_h / (4.0*pow(np.pi, 2))  # E, White PM, tau^-1
         elif self.b == -1:
-            coeff = (1.038+3*np.log(2.0*np.pi*f_h*tau))/(4.0*pow(np.pi, 2))  # D, Flicker PM, tau^-1
+            # D, Flicker PM, tau^-1
+            coeff = (1.038+3*np.log(2.0*np.pi*f_h*tau))/(4.0*pow(np.pi, 2))
         elif self.b == -2:
             coeff = 0.5  # C, white FM,  1/sqrt(tau)
         elif self.b == -3:
@@ -286,13 +287,17 @@ class Noise(object):
         if self.b == 0:
             coeff = 3.0/(8.0*pow(np.pi, 2))  # E, White PM, tau^-{3/2}
         elif self.b == -1:
-            coeff = (24.0*np.log(2)-9.0*np.log(3))/8.0/pow(np.pi, 2)  # D, Flicker PM, tau^-1
+            # D, Flicker PM, tau^-1
+            coeff = (24.0*np.log(2)-9.0*np.log(3))/8.0/pow(np.pi, 2)
         elif self.b == -2:
-            coeff = 0.25  # C, white FM,  1/sqrt(tau)
+            # C, white FM,  1/sqrt(tau)
+            coeff = 0.25
         elif self.b == -3:
-            coeff = 2.0*np.log(3.0*pow(3.0, 11.0/16.0)/4.0)  # B, flicker FM,  constant MDEV
+            # B, flicker FM,  constant MDEV
+            coeff = 2.0*np.log(3.0*pow(3.0, 11.0/16.0)/4.0)
         elif self.b == -4:
-            coeff = 11.0/20.0*pow(np.pi, 2)  # A, RW FM, sqrt(tau)
+            # A, RW FM, sqrt(tau)
+            coeff = 11.0/20.0*pow(np.pi, 2)
 
         return np.sqrt(coeff*g_b*pow(2.0*np.pi, 2))
 

--- a/allantools/plot.py
+++ b/allantools/plot.py
@@ -66,8 +66,7 @@ class Plot(object):
     def plot(self, atDataset,
              errorbars=False,
              grid=False,
-             **kwargs
-            ):
+             **kwargs):
         """ Use matplotlib methods for plotting
 
         Additional keywords arguments are passed to
@@ -87,13 +86,11 @@ class Plot(object):
             self.ax.errorbar(atDataset.out["taus"],
                              atDataset.out["stat"],
                              yerr=atDataset.out["stat_err"],
-                             **kwargs
-                            )
+                             **kwargs)
         else:
             self.ax.plot(atDataset.out["taus"],
                          atDataset.out["stat"],
-                         **kwargs
-                        )
+                         **kwargs)
         self.ax.set_xlabel("Tau")
         self.ax.set_ylabel(atDataset.out["stat_id"])
         self.ax.grid(grid, which="minor", ls="-", color='0.65')

--- a/allantools/realtime.py
+++ b/allantools/realtime.py
@@ -41,7 +41,8 @@ class dev_realtime(object):
         self.auto_afs = auto_afs
         # logscpace will fail at >=6 (?), need to remove duplicates?
         self.af_taus = numpy.logspace(0, 1, pts_per_decade+1)[:-1]
-        self.af_idx = 0       # logspace index, to keep track of afs in auto-af mode
+        # logspace index, to keep track of afs in auto-af mode
+        self.af_idx = 0
         self.af_decade = 0    # logspace decade
         if auto_afs:
             self.afs = numpy.array([1])
@@ -59,11 +60,14 @@ class dev_realtime(object):
             next_idx = 0
             next_decade = self.af_decade + 1
 
-        next_af = int(numpy.round(pow(10.0, next_decade) * self.af_taus[next_idx]))  # next possible AF
+        # next possible AF:
+        next_af = int(numpy.round(
+            pow(10.0, next_decade) * self.af_taus[next_idx]))
         if len(self.x) >= (2*next_af+1):  # can compute next AF
             self.afs = numpy.append(self.afs, next_af)  # new AF
             self.add_af()  # tell subclass to update internal variables
-            # self.S = numpy.append(self.S, 0) # new S, FIXME: S defined in subclass!
+            # FIXME: S defined in subclass!
+            # self.S = numpy.append(self.S, 0) # new S
             self.dev = numpy.append(self.dev, 0)  # new dev
             self.af_idx = next_idx
             self.af_decade = next_decade
@@ -90,13 +94,16 @@ class dev_realtime(object):
 
 
 class oadev_realtime(dev_realtime):
-    """ Overlapping Allan deviation in real-time from a stream of phase/frequency samples.
+    """ Overlapping Allan deviation in real-time from a stream of
+    phase/frequency samples.
 
-        Dobrogowski & Kasznia
-        https://doi.org/10.1109/FREQ.2007.4319204
+    Dobrogowski & Kasznia
+    https://doi.org/10.1109/FREQ.2007.4319204
     """
     def __init__(self, afs=[1], tau0=1.0, auto_afs=False, pts_per_decade=4):
-        super(oadev_realtime, self).__init__(afs=afs, tau0=tau0, auto_afs=auto_afs, pts_per_decade=pts_per_decade)
+        super(oadev_realtime, self).__init__(afs=afs, tau0=tau0,
+                                             auto_afs=auto_afs,
+                                             pts_per_decade=pts_per_decade)
         self.S = numpy.zeros(len(afs))      # sum-of-squares
 
     def add_phase(self, xnew):
@@ -114,21 +121,25 @@ class oadev_realtime(dev_realtime):
         i = len(self.x)-1  # last pt
         S_new = pow(self.x[i] - 2*self.x[i-af] + self.x[i-2*af], 2)
         self.S[idx] = self.S[idx] + S_new
-        self.dev[idx] = numpy.sqrt((1.0/(2*pow(af*self.tau0, 2)*(i+1-2*af))) * self.S[idx])
+        self.dev[idx] = numpy.sqrt(
+            (1.0/(2*pow(af*self.tau0, 2)*(i+1-2*af))) * self.S[idx])
 
     def add_af(self):
         self.S = numpy.append(self.S, 0)
 
 
 class ohdev_realtime(dev_realtime):
-    """ Overlapping Hadamard deviation in real-time from a stream of phase/frequency samples.
+    """ Overlapping Hadamard deviation in real-time from a stream of
+    phase/frequency samples.
 
-        [Dobrogowski2007]_
-        Dobrogowski & Kasznia
-        https://doi.org/10.1109/FREQ.2007.4319204
+    [Dobrogowski2007]_
+    Dobrogowski & Kasznia
+    https://doi.org/10.1109/FREQ.2007.4319204
     """
     def __init__(self, afs=[1], tau0=1.0, auto_afs=False, pts_per_decade=4):
-        super(ohdev_realtime, self).__init__(afs=afs, tau0=tau0, auto_afs=auto_afs, pts_per_decade=pts_per_decade)
+        super(ohdev_realtime, self).__init__(afs=afs, tau0=tau0,
+                                             auto_afs=auto_afs,
+                                             pts_per_decade=pts_per_decade)
         self.S = numpy.zeros(len(afs))      # sum-of-squares
 
     def add_af(self):
@@ -148,19 +159,26 @@ class ohdev_realtime(dev_realtime):
         af = self.afs[idx]
         i = len(self.x)-1  # last pt
         # print i,self.x
-        S_new = pow(self.x[i] - 3*self.x[i-af] + 3*self.x[i-2*af] - self.x[i-3*af], 2)
+        S_new = pow(self.x[i] -
+                    3*self.x[i-af] +
+                    3*self.x[i-2*af] -
+                    self.x[i-3*af], 2)
         self.S[idx] = self.S[idx] + S_new
-        self.dev[idx] = numpy.sqrt((1.0/(6.0*pow(af*self.tau0, 2)*(i+1.0-3*af))) * self.S[idx])
+        self.dev[idx] = numpy.sqrt(
+            (1.0/(6.0*pow(af*self.tau0, 2)*(i+1.0-3*af))) * self.S[idx])
 
 
 class tdev_realtime(dev_realtime):
-    """ Time deviation and Modified Allan deviation in real-time from a stream of phase/frequency samples.
+    """ Time deviation and Modified Allan deviation in real-time from a stream
+    of phase/frequency samples.
 
-        Dobrogowski & Kasznia
-        https://doi.org/10.1109/FREQ.2007.4319204
+    Dobrogowski & Kasznia
+    https://doi.org/10.1109/FREQ.2007.4319204
     """
     def __init__(self, afs=[1], tau0=1.0, auto_afs=False, pts_per_decade=4):
-        super(tdev_realtime, self).__init__(afs=afs, tau0=tau0, auto_afs=auto_afs, pts_per_decade=pts_per_decade)
+        super(tdev_realtime, self).__init__(afs=afs, tau0=tau0,
+                                            auto_afs=auto_afs,
+                                            pts_per_decade=pts_per_decade)
         self.S = numpy.zeros(len(afs))      # sum-of-squares
         self.So = numpy.zeros(len(afs))      # overall sum-of-squares
 
@@ -194,7 +212,8 @@ class tdev_realtime(dev_realtime):
         # Eqn (14)
         num_pts = len(self.x)
         af = self.afs[idx]
-        self.dev[idx] = numpy.sqrt((1.0/6.0)*(1.0/(num_pts-3*af+1.0))*(1.0/pow(af, 2))*(self.So[idx]))
+        self.dev[idx] = numpy.sqrt(
+            (1.0/6.0)*(1.0/(num_pts-3*af+1.0))*(1.0/pow(af, 2))*(self.So[idx]))
 
     def update_S(self, idx):
         """ update S, sum-of-squares """
@@ -202,11 +221,13 @@ class tdev_realtime(dev_realtime):
         assert(len(self.x) >= 3*af+1)
         i = len(self.x)-1  # last pt
         # Eqn (12)
-        S_new = -1*self.x[i-3*af] + 3*self.x[i-2*af] - 3*self.x[i-af] + self.x[i]
+        S_new = (-1*self.x[i-3*af] + 3*self.x[i-2*af] -
+                 3*self.x[i-af] + self.x[i])
 
         self.S[idx] = self.S[idx] + S_new
         # Eqn (11)
-        self.So[idx] = self.So[idx] + pow(self.S[idx], 2)  # ??? S_(i-1) in paper for TDEV-sqrt?
+        self.So[idx] = self.So[idx] + pow(self.S[idx], 2)
+        # ??? S_(i-1) in paper for TDEV-sqrt?
         self.update_dev(idx)
 
     def mdev(self):

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,0 +1,3 @@
+[pytest]
+markers = 
+	slow: marks tests as slow (deselect with '-m "not slow"')

--- a/tests/functional_tests/test_plot.py
+++ b/tests/functional_tests/test_plot.py
@@ -2,9 +2,7 @@
 
 import sys
 sys.path.append("..")
-
 import allantools as at
-import numpy
 import pytest
 
 
@@ -12,8 +10,8 @@ def test_plot():
     ds = at.Dataset(data=at.noise.white(1000), rate=1.234)
     ds.compute("adev")
     p = at.Plot(no_display=True)
-    p.plot(ds)
+    p.plot(ds, errorbars=True)
     p.show()
-    
+
 if __name__ == "__main__":
     test_plot()

--- a/tests/phasedat/test_phasedat.py
+++ b/tests/phasedat/test_phasedat.py
@@ -8,12 +8,12 @@
 import sys
 import pytest
 import allantools as allan
-import testutils
 import os
 
 sys.path.append("..")
 sys.path.append("../..")  # hack to import from parent directory
 # remove if you have allantools installed in your python path
+import testutils
 
 
 data_file = 'PHASE.DAT'
@@ -75,6 +75,11 @@ class TestPhaseDat():
     def test_phasedat_mtie(self):
         self.generic_test(result='phase_dat_mtie.txt',
                           fct=allan.mtie)
+
+    @pytest.mark.xfail(reason="mtie_phase_fast not finished")
+    def test_phasedat_mtie_phase_fast(self):
+        self.generic_test(result='phase_dat_mtie.txt',
+                          fct=allan.mtie_phase_fast)
 
     def test_phasedat_tierms(self):
         self.generic_test(result='phase_dat_tierms.txt',

--- a/tests/phasedat/test_phasedat.py
+++ b/tests/phasedat/test_phasedat.py
@@ -5,23 +5,21 @@
   PHASE.DAT comes with Stable32 (version 1.53 was used in this case)
 
 """
-
-import math
 import sys
 import pytest
-
-sys.path.append("..")
-sys.path.append("../..") # hack to import from parent directory
-# remove if you have allantools installed in your python path
-
 import allantools as allan
 import testutils
-
 import os
+
+sys.path.append("..")
+sys.path.append("../..")  # hack to import from parent directory
+# remove if you have allantools installed in your python path
+
 
 data_file = 'PHASE.DAT'
 tolerance = 1e-4
 verbose = 1
+
 
 def change_to_test_dir():
     # hack to run script from its own directory
@@ -29,55 +27,67 @@ def change_to_test_dir():
     dname = os.path.dirname(abspath)
     os.chdir(dname)
 
+
 class TestPhaseDat():
     def test_phasedat_adev(self):
-        self.generic_test( result= 'phase_dat_adev.txt' , fct= allan.adev, verbose=True )
-        
+        self.generic_test(result='phase_dat_adev.txt',
+                          fct=allan.adev, verbose=True)
+
     def test_phasedat_oadev(self):
-        self.generic_test( result='phase_dat_oadev.txt' , fct= allan.oadev )
-        
+        self.generic_test(result='phase_dat_oadev.txt', fct=allan.oadev)
+
     def test_phasedat_mdev(self):
-        self.generic_test( result='phase_dat_mdev.txt' , fct= allan.mdev )
-        
+        self.generic_test(result='phase_dat_mdev.txt', fct=allan.mdev)
+
     def test_phasedat_tdev(self):
-        self.generic_test( result='phase_dat_tdev.txt' , fct= allan.tdev )
-        
+        self.generic_test(result='phase_dat_tdev.txt', fct=allan.tdev)
+
     def test_phasedat_hdev(self):
-        self.generic_test( result='phase_dat_hdev.txt' , fct= allan.hdev )
-        
+        self.generic_test(result='phase_dat_hdev.txt', fct=allan.hdev)
+
     def test_phasedat_ohdev(self):
-        self.generic_test( result='phase_dat_ohdev.txt' , fct= allan.ohdev )
-        
+        self.generic_test(result='phase_dat_ohdev.txt', fct=allan.ohdev)
+
     def test_phasedat_totdev(self):
-        self.generic_test( result='phase_dat_totdev.txt' , fct= allan.totdev )
-        
+        self.generic_test(result='phase_dat_totdev.txt', fct=allan.totdev)
+
     @pytest.mark.slow
     def test_phasedat_htotdev(self):
-        # bias-correction is not implemented, so compare against Stable32-run without bias correction
-        self.generic_test( result='phase_dat_htotdev_octave_nobias.txt' , fct= allan.htotdev )
-        
+        # bias-correction is not implemented, so compare against Stable32-run
+        # without bias correction
+        self.generic_test(result='phase_dat_htotdev_octave_nobias.txt',
+                          fct=allan.htotdev)
+
     @pytest.mark.slow
     def test_phasedat_mtotdev(self):
-        self.generic_test( result='phase_dat_mtotdev_octave.txt' , fct= allan.mtotdev )
-        
+        self.generic_test(result='phase_dat_mtotdev_octave.txt',
+                          fct=allan.mtotdev)
+
     @pytest.mark.slow
     def test_phasedat_ttotdev(self):
-        self.generic_test( result='phase_dat_ttotdev_octave.txt' , fct= allan.ttotdev )
+        self.generic_test(result='phase_dat_ttotdev_octave.txt',
+                          fct=allan.ttotdev)
 
     def test_phasedat_theo1_alpha0(self):
-        self.generic_test( result='phase_dat_theo1_alpha0_decade.txt' , fct= allan.theo1 )
-        
+        self.generic_test(result='phase_dat_theo1_alpha0_decade.txt',
+                          fct=allan.theo1)
+
     def test_phasedat_mtie(self):
-        self.generic_test( result='phase_dat_mtie.txt' , fct= allan.mtie )
-        
+        self.generic_test(result='phase_dat_mtie.txt',
+                          fct=allan.mtie)
+
     def test_phasedat_tierms(self):
-        self.generic_test( result='phase_dat_tierms.txt' , fct= allan.tierms )
-    
-    def generic_test(self, datafile = data_file, result="", fct=None, verbose=False):
+        self.generic_test(result='phase_dat_tierms.txt',
+                          fct=allan.tierms)
+
+    def generic_test(self, datafile=data_file, result="",
+                     fct=None, verbose=False):
         change_to_test_dir()
-        testutils.test_row_by_row( fct, datafile, 1.0, result , verbose=verbose, tolerance=tolerance)
+        testutils.test_row_by_row(fct, datafile, 1.0, result,
+                                  verbose=verbose, tolerance=tolerance)
+
 
 if __name__ == "__main__":
-    #pytest.main()
+    # pytest.main()
     t = TestPhaseDat()
     t.test_phasedat_adev()

